### PR TITLE
bump to libCEC 4.0.0 + some rebased PRs

### PR DESCRIPTION
**Pull request by opdenkamp**
Changes for the libCEC 4.0.0 bump and rebased PRs #8116 and #8549, so we don't have to wait for these to be fixed up again.

It's a major bump because of some interface changes. Mainly cleaned up old stuff that's no longer being used.

libCEC hasn't been tagged yet, just putting up the PR now for review and testing. libCEC's 4.0.0 "beta" is in the master branch

edit: fixed wrong pr linked (missing number)

### CODE REVIEW

File: system/peripherals.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <setting key="send_inactive_source" type="bool" value="1" label="36025" order="5" />
     <setting key="cec_standby_screensaver" type="bool" value="0" label="36009" order="6" />
     <setting key="cec_wake_screensaver" type="bool" value="1" label="36010" order="7" />
-    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011" />
+    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011|13009|36043|36045" />
```
Comment (popcornmix) [inline system/peripherals.xml]: 36043 is "Number of buffers used by graphics driver" which doesn't sound right here

Comment (opdenkamp) [inline system/peripherals.xml]: fixed. the code had the correct id, the xml didn't. thanks

**Comment by wsnipex** [comment]
please bump version in https://github.com/xbmc/xbmc/blob/master/project/cmake/modules/FindCEC.cmake#L18

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp [hunk 2]
```diff
@@ -614,43 +623,67 @@ void CPeripheralCecAdapter::SetMenuLanguage(const char *strLanguage)
...
+  case LOCALISED_ID_SUSPEND:
+    m_bStarted = false;
+    g_application.ExecuteXBMCAction("Suspend");
+    break;
+  case LOCALISED_ID_QUIT:
+    m_bStarted = false;
+    g_application.ExecuteXBMCAction("Quit");
+    break;
+  case LOCALISED_ID_PAUSE:
+    m_bStarted = false;
```
Comment (Glenn-1990) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: Pause does not quit kodi or libcec, so this should be handled as stop below?

Comment (opdenkamp) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: yup, you're right, thanks

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp [hunk 3]
```diff
@@ -1385,6 +1385,7 @@ void CPeripheralCecAdapter::SetConfigurationFromSettings(void)
   m_bPowerOffScreensaver               = GetSettingBool("cec_standby_screensaver") ? 1 : 0;
   m_bPowerOnScreensaver                = GetSettingBool("cec_wake_screensaver") ? 1 : 0;
   m_bSendInactiveSource                = GetSettingBool("send_inactive_source") ? 1 : 0;
+  m_configuration.bAutoWakeAVR         = GetSettingBool("power_avr_on_as") ? 1 : 0;
```
Comment (Glenn-1990) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: There is no SetSetting for this in SetConfigurationFromLibCEC(), not sure if that's really needed?

Comment (opdenkamp) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: It is. This is read from Kodi's config only. I got rid of all the unneeded stuff in `SetConfigurationFromLibCEC()`

**Comment by opdenkamp** [comment]
@wsnipex I've uploaded the debs to https://launchpad.net/~pulse-eight/+archive/ubuntu/libcec and updated FindCEC.cmake

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp (hunk 2 above)
Comment (opdenkamp) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: fixed now

**Comment by wsnipex** [comment]
thanks, added it to our PPA as well

**Comment by opdenkamp** [comment]
jenkins build this please

**Comment by opdenkamp** [comment]
looks like the internal libCEC build from depends isn't happy yet. will check later this weekend

**Comment by opdenkamp** [comment]
depends are building fine here and the build log is gone. jenkins build this please

**Comment by opdenkamp** [comment]
platform lib in depends needs a bump too, will fix it

**Comment by opdenkamp** [comment]
jenkins build this please

**Comment by opdenkamp** [comment]
no idea what this travis thing thinks has failed, but jenkins is happy with it, so I'll clean this up and it's ready for merging then afaict

**Comment by hudokkow** [comment]
Don't mind Travis-CI. It fails on a test with clang
> The following tests FAILED:
	393 - TestThreadLocal.Stack (SEGFAULT)

**Comment by opdenkamp** [comment]
thanks @hudokkow 

PR rebased onto master, label id updated cause something else took it and squashed commits, so it's ready if jenkins is happy with it, so jenkins build this please

File: system/peripherals.xml [hunk 4]
```diff
@@ -16,15 +16,15 @@
...
+    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011|13009|36044|36046" />
     <setting key="standby_tv_on_pc_standby" type="bool" value="1" label="36026" order="9" />
     <setting key="use_tv_menu_language" type="bool" value="1" label="36018" order="10" />
     <setting key="pause_playback_on_deactivate" type="bool" value="1" label="36033" configurable="0" />
     <setting key="pause_or_stop_playback_on_deactivate" type="enum" value="231" label="36033" order="11" lvalues="231|36044|36045" />
     <setting key="connected_device" type="enum" label="36019" value="36037" lvalues="36037|36038" order="12" />
     <setting key="cec_hdmi_port" type="int" value="1" min="1" max="15" label="36015" order="13" />
     <setting key="physical_address" type="string" label="36021" value="0" order="14" />
-    <setting key="port" type="string" value="" label="36022" order="15" />
+    <setting key="power_avr_on_as" type="bool" label="36045" value="0" order="15" />
```
Comment (Rechi) [inline system/peripherals.xml]: shouldn't this be `#36046`?

File: system/peripherals.xml [hunk 5]
```diff
@@ -16,15 +16,15 @@
     <setting key="send_inactive_source" type="bool" value="1" label="36025" order="5" />
     <setting key="cec_standby_screensaver" type="bool" value="0" label="36009" order="6" />
     <setting key="cec_wake_screensaver" type="bool" value="1" label="36010" order="7" />
-    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011" />
+    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011|13009|36044|36046" />
```
Comment (Rechi) [inline system/peripherals.xml]: shouldn't this be `#36045` instead of `#36046`?

File: system/peripherals.xml (hunk 4 above)
Comment (opdenkamp) [inline system/peripherals.xml]: meeh damn magic numbers... you're right, will fix it tonight

Comment (ksooo) [inline system/peripherals.xml]: Ping @opdenkamp. You wanted to fix this. ;-)

**Pull request merged**